src/edge_utils.c

```diff
@@ -2055,8 +2055,8 @@ void readFromIPSocket(n2n_edge_t * eee, int in_sock) {
                     traceEvent(TRACE_NORMAL, "Supernode '%s' added to the list of supernodes.", sn->ip_addr);
                   }
 
-                  tmp_sock = (void*)tmp_sock + REG_SUPER_ACK_PAYLOAD_ENTRY_SIZE;
-                  tmp_mac = (void*)tmp_sock + sizeof(n2n_sock_t);
+                  tmp_sock = (char*)tmp_sock + REG_SUPER_ACK_PAYLOAD_ENTRY_SIZE;
```

Comment:
ref  :  added payload struct to carry supernode data #501
The  #501 might be a better fix.

Reply:
please, merge #501 first, after that, I will follow the change from main dev branch.